### ISSUE TITLE
Update xUnit1041 to support generic fixtures

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2886]
Using the generic collection fixture feature enabled by #2557 currently triggers xUnit1041.

```csharp
using Xunit;

public class GenericClass<T1, T2>
{ }

[CollectionDefinition("GenericFixture")]
public class GenericFixture<T1, T2> : ICollectionFixture<GenericClass<T1, T2>>
{ }

[Collection("GenericFixture")]
public class TestClass
{
    public TestClass(GenericClass<string, int> fixture)
    { }

    [Fact]
    public void TestMethod()
    { }
}
```

![image](https://github.com/xunit/xunit/assets/16944/3e5c704e-6097-43e9-a621-a17b05af95c3)

**Comment by JamesTerwilliger** [comment]
Fix present here: https://github.com/xunit/xunit.analyzers/pull/180

**Comment by bradwilson** [comment]
Available in `1.11.0-pre.10` https://xunit.net/docs/using-ci-builds
